src/runtime/set.cpp

```diff
@@ -489,10 +483,9 @@ Box* setHash(BoxedSet* self) {
     RELEASE_ASSERT(isSubclass(self->cls, frozenset_cls), "");
 
     int64_t rtn = 1927868237L;
-    for (Box* e : self->s) {
-        BoxedInt* h = hash(e);
-        assert(PyInt_Check(h));
-        rtn ^= h->n + 0x9e3779b9 + (rtn << 6) + (rtn >> 2);
+    for (auto&& e : self->s) {
+        assert(e.hash == unboxInt(hash(e.value)));
```

Comment:
I think we can take this out -- looks like CPython will use the stored hash value as well.

otherwise lgtm!